app/acknowledgement/process-invalid.js

```diff
@@ -3,14 +3,20 @@ const { getHoldCategoryId } = require('../holds')
 const getHoldCategoryName = require('./get-hold-category-name')
 const holdAndReschedule = require('../reschedule')
 const { resetPaymentRequestById } = require('../reset')
+const { sendAcknowledgementEvent } = require('../event')
 
-const processInvalid = async (schemeId, paymentRequestId, frn, message) => {
+const config = require('../config')
+
+const processInvalid = async (schemeId, paymentRequestId, frn, acknowledgement) => {
   const transaction = await db.sequelize.transaction()
   try {
     await resetPaymentRequestById(paymentRequestId, schemeId, transaction)
-    const holdCategoryName = getHoldCategoryName(message)
+    const holdCategoryName = getHoldCategoryName(acknowledgement.message)
     const holdCategoryId = await getHoldCategoryId(schemeId, holdCategoryName, transaction)
     await holdAndReschedule(schemeId, paymentRequestId, holdCategoryId, frn, transaction)
+    if (config.isAlerting && !acknowledgement.success) {
```

Comment:
`&& !acknowledgement.success` is not needed as `processInvalid` is only called in `update-acknowledgement` when `!acknowledgement.success`

Reply:
I added that validation in because one of the tests I wrote would fail without it, I can try and re-write the test though if you think this bit of code should be omitted?

Reply:
I see the testing question, `should not call sendAcknowledgementEvent when a successful ack object is given and isAlerting is true`.

Question is, is this possible to be the case given the function's expectation?

Counterargument could be it is a case of defence in depth.

Reply:
Happy to leave this decision to yourself

Reply:
Sounds good!